app/src/main/java/com/openvehicles/OVMS/entities/CarData.kt

```diff
@@ -693,6 +739,19 @@ class CarData : Serializable {
                 } else {
                     ""
                 }
+                car_charge_kwh_grid = dataParts[38].toFloat()
+                car_charge_kwh_grid_total = dataParts[39].toFloat()
+                car_battery_capacity = dataParts[40].toFloat()
```

Comment:
This is a crash recipe. You need to check the array length before accessing elements, as not all firmware versions installed on user modules already send these new elements. In this case the solution is quite simple, as you already added the `if size >= 42` in the following block, you can simply move these three lines into that if block.